### PR TITLE
feat: add support for multiple + glob secrets

### LINKED ISSUE #170: Feat: Support multiple secret files
Could you add a feature that either accepts a manually specified list of paths in `config.yml` for multiple secret files to read or that simply looks at `config/secrets/*.yml` instead of a singular `secrets.yml`?

My use case is that I'm using Kubernetes with separate secrets containing the API keys for each of my *arrs, so I can't mount one singular secret into the container as a file and give Configarr everything it needs. For now I'm using `!env` as a workaround, but I'd really rather use the `!secret` option.

### PR DESCRIPTION
**Pull request by BlackDark**
Closes #170

## Summary by Sourcery

Add support for loading secrets from multiple paths and glob patterns while maintaining backward compatibility with a single secrets file.

New Features:
- Support loading and merging secrets from multiple files specified via glob patterns or comma-separated paths.
- Cache loaded secrets to avoid repeated file reads within a process.

Enhancements:
- Gracefully handle missing, empty, or invalid secret files with logging and fall back to empty secrets when appropriate.

Build:
- Add fast-glob as a runtime dependency for expanding secret file patterns.

Tests:
- Add comprehensive tests covering single-file, glob, and comma-separated secret configurations, including caching, merge precedence, and error/edge cases.

### CODE REVIEW

File: src/config.test.ts [hunk 1]
```diff
@@ -668,3 +691,399 @@ describe("custom_formats ordering", () => {
...
+    expect(secrets1).toBe(secrets2);
+    // Should only read file once (cached on second call)
+    expect(mockReadFileSync).toHaveBeenCalledTimes(1);
+  });
+
+  test("should load multiple secret files via comma-separated list", () => {
+    const secretLocation = "/config/secrets1.yml,/config/secrets2.yml";
+    const secretFile1 = "/config/secrets1.yml";
+    const secretFile2 = "/config/secrets2.yml";
+    const secretContent1 = "SONARR_API_KEY: sonarr-key-123";
```
[the remark this answers is not in the document]
Comment (BlackDark) [reply src/config.test.ts]: no leak

File: src/config.test.ts [hunk 2]
```diff
@@ -668,3 +691,399 @@ describe("custom_formats ordering", () => {
...
+    mockReadFileSync.mockImplementation((path) => {
+      if (path === secretFile1) return secretContent1;
+      if (path === secretFile2) return secretContent2;
+      return "";
+    });
+
+    const secrets = getSecrets();
+
+    expect(secrets).toEqual({
+      SONARR_API_KEY: "sonarr-key-123",
```
[the remark this answers is not in the document]
Comment (BlackDark) [reply src/config.test.ts]: no leak
